### PR TITLE
Test/ Transaction status modal

### PR DESCRIPTION
**Pull request by w1that**
# Add comprehensive test coverage for Transaction Status Modal utilities

### Key additions:
- Complete test coverage for `getFormattedType` including verb variations
- Comprehensive tests for `getTransactionStatusModalMessage` covering all transaction states
- Test cases for `getTransactionStatusModalTitle` with various scenarios
- Validation of `getTransactionStatusModalSpinnerTitle` for all status types

### Test coverage includes:
- Transaction type formatting (transfer, listing, buy, sell, cancel, offer)
- Status-specific messages and titles
- Edge cases like undefined transaction types
- Special handling for offers with price information
- Order ID scenarios
- Timeout and failure states

# Add integration tests for TransactionStatusModal component
### Key test coverage:
- Modal visibility control (show/hide)
- Transaction state transitions:
  - Processing state display
  - Success state handling
  - Failure state management
  - Timeout state handling
- Callback execution validation:
  - Success callback triggers
  - Error callback handling

### Test scenarios include:
- Initial closed state verification
- Processing state message and UI elements
- Success state with blockchain confirmation
- Error handling with transaction failures
- Timeout handling for long-running transactions
- Callback execution with transaction results

### CODE REVIEW

File: packages/sdk/src/react/ui/modals/_internal/components/transactionStatusModal/__tests__/TransactionStatusModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+	it('should show success state when transaction is successful', async () => {
+		mockPublicClient.waitForTransactionReceipt.mockResolvedValue({
+			status: 'success',
+		});
+		transactionStatusModal$.open(mockTransactionArgs);
+		render(<TransactionStatusModal />);
+
+		await waitFor(() => {
+			const elements = screen.queryAllByTestId('transaction-status-title');
```
Comment (AlexanderKolberg) [inline packages/sdk/src/react/ui/modals/_internal/components/transactionStatusModal/__tests__/TransactionStatusModal.test.tsx:73]: Why is there multiple elements with the same test ID in the DOM here? That sound like a bug?

Comment (w1that) [reply packages/sdk/src/react/ui/modals/_internal/components/transactionStatusModal/__tests__/TransactionStatusModal.test.tsx]: Nice catch, fixed on https://github.com/0xsequence/marketplace-sdk/pull/153/commits/6a5d1e56a76c67638a10b3f0031ec41fa9577c3d

Before it, I ran added the following to make sure there is only one element at the time:
```
expect(elements).toHaveLength(1); // passes ✅
```

**Review by AlexanderKolberg — APPROVED** [review]
LGTM
